tests/integration/steps.test.js

```diff
@@ -62,74 +65,82 @@ describe('STEPS ROUTES - PUT /steps/{_id}', () => {
       expect(stepUpdated).toEqual(expect.objectContaining({ _id: stepId, activities: payload.activities }));
     });
 
-    it('should return a 400 if payload is empty', async () => {
+    it('should return a 404 if step is invalid', async () => {
       const response = await app.inject({
         method: 'PUT',
-        url: `/steps/${stepId.toHexString()}`,
-        payload: {},
+        url: `/steps/${new ObjectID()}`,
+        payload: { name: 'une nouvelle étape super innovante' },
         headers: { Cookie: `alenvi_token=${authToken}` },
       });
 
-      expect(response.statusCode).toBe(400);
+      expect(response.statusCode).toBe(404);
     });
 
-    it('should return a 400 if name is equal to \'\' ', async () => {
+    it('should return a 403 if step is published and payload has activities', async () => {
+      const payload = { activities: [stepsList[0].activities[1], stepsList[0].activities[0]], name: 'skusku' };
       const response = await app.inject({
         method: 'PUT',
-        url: `/steps/${stepId.toHexString()}`,
-        payload: { name: '' },
+        url: `/steps/${stepsList[3]._id}`,
+        payload,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(response.statusCode).toBe(403);
+    });
+
+    it('should return a 400 if payload is empty', async () => {
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/steps/${stepId}`,
+        payload: {},
         headers: { Cookie: `alenvi_token=${authToken}` },
       });
 
       expect(response.statusCode).toBe(400);
     });
 
-    it('should return a 400 if lengths are not equal', async () => {
-      const payload = { activities: [stepsList[0].activities[1]] };
+    it('should return a 400 if name is empty ', async () => {
       const response = await app.inject({
         method: 'PUT',
-        url: `/steps/${stepId.toHexString()}`,
-        payload,
+        url: `/steps/${stepId}`,
+        payload: { name: '' },
         headers: { Cookie: `alenvi_token=${authToken}` },
       });
 
       expect(response.statusCode).toBe(400);
     });
 
-    it('should return a 400 if actvities from payload and from db are not the same', async () => {
-      const payload = { activities: [stepsList[0].activities[1], new ObjectID()] };
+    it('should return a 400 if activities\' lengths from db and payload are not equal', async () => {
```

Comment:
Est-ce que ce test est vraiment utile? J'ai l'impression qu'il est inclus dans le test `should return a 400 if activities from payload and from db are not strict equal`

Reply:
le nom du test est peut-être un peu confus. Il ne test pas la strict égalité car la strict égalité c'est : les chaque array contient l'autre ou bien une array contient l'autre et elles ont la même taille

Donc il faut bien les deux tests je dirais